tests/test_decoding/test_decoder_properties.py

```diff
@@ -183,24 +185,87 @@ def test_decode_signed_int(integer_bit_size, stream_bytes, data_byte_size):
 
 @settings(max_examples=250)
 @given(
-    string_bytes=st.binary(min_size=0, max_size=256),
+    _bytes=st.binary(min_size=0, max_size=256),
     pad_size=st.integers(min_value=0, max_value=32),
 )
-def test_decode_bytes_and_string(string_bytes, pad_size):
-    size_bytes = zpad32(int_to_big_endian(len(string_bytes)))
-    padded_string_bytes = string_bytes + b'\x00' * pad_size
-    stream_bytes = size_bytes + padded_string_bytes
+def test_decode_bytes(_bytes, pad_size):
+    size_bytes = zpad32(int_to_big_endian(len(_bytes)))
+    padded_bytes = _bytes + b'\x00' * pad_size
+    stream_bytes = size_bytes + padded_bytes
+    stream = ContextFramesBytesIO(stream_bytes)
+
+    decoder = ByteStringDecoder()
+
+    if len(padded_bytes) < ceil32(len(_bytes)):
+        with pytest.raises(InsufficientDataBytes):
+            decoder(stream)
+        return
+
+    decoded_value = decoder(stream)
+    assert decoded_value == _bytes
+
+
+@settings(max_examples=250)
+@given(
+    _strings=st.text(min_size=0, max_size=256),
+    pad_size=st.integers(min_value=0, max_value=32),
+)
+def test_decode_strings(_strings, pad_size):
+    st.assume(is_utf8_encodable(_strings))
+    size_bytes = zpad32(int_to_big_endian(len(_strings.encode("utf-8"))))
+    padded_bytes = _strings.encode("utf-8") + b'\x00' * pad_size
+    stream_bytes = size_bytes + padded_bytes
     stream = ContextFramesBytesIO(stream_bytes)
 
     decoder = StringDecoder()
 
-    if len(padded_string_bytes) < ceil32(len(string_bytes)):
+    if len(padded_bytes) < ceil32(len(_strings.encode("utf-8"))):
         with pytest.raises(InsufficientDataBytes):
             decoder(stream)
         return
 
     decoded_value = decoder(stream)
-    assert decoded_value == string_bytes
+    assert decoded_value == _strings
+
+
+def is_utf8_decodable(value):
+    try:
+        value.decode("utf-8")
+    except UnicodeDecodeError:
+        return False
+    return True
+
+
+def is_utf8_encodable(value):
+    try:
+        value.encode("utf-8")
+    except UnicodeEncodeError:
+        return False
+    return True
+
+
+@settings(max_examples=250)
+@given(
+    _bytes=st.binary(min_size=0, max_size=256),
+    pad_size=st.integers(min_value=0, max_value=32),
+)
+def test_decode_strings_raises(_bytes, pad_size):
+    size_bytes = zpad32(int_to_big_endian(len(_bytes)))
+    padded_bytes = _bytes + b'\x00' * pad_size
+    stream_bytes = size_bytes + padded_bytes
+    stream = ContextFramesBytesIO(stream_bytes)
+
+    decoder = StringDecoder()
+
+    st.assume(not is_utf8_decodable(_bytes))
```

Comment:
Instead of this you can do a `.filter()` call on the bytes strategy.

```python
from cytoolz import compliment

@given(_bytes=st.binary(...).filter(compliment(is_utf8_decodable)), ...)
```

This should only give you byte strings which are not utf8 decodable.